HealthcareSystem/Backend/Service/RoomAndEquipment/RoomService.cs

```diff
@@ -93,5 +104,51 @@ public List<Room> ViewRoomByUsage(TypeOfUsage usage, DateTime beginDate, DateTim
             return result;
         }
 
+        public ICollection<Room> GetRoomsByUsageAndEquipment(TypeOfUsage usage, ICollection<int> requiredEquipmentTypeIds)
```

Comment:
I feel like this could be implemented in a much simpler way, maybe through a nice database query with some joins, or just though using some fields (although, again, that might lead to some clunky Linq expressions). However, when I went to look for better ways to implement this, here's what I noticed - why isn't EquipmentInRooms annotated with foreign keys at all? I see that the correct key is set in OnModelCreating, but it does not have the correct foreign key constraints.

Reply:
As database query is concerned, that is what I thought of also, but the expressions would really look nasty and a bit unreadable. I will try to think of maybe 2 separate queries that would make it shorter and clearer, but it will need to wait for now.
As foreign keys are concerned, I am not really sure why there aren't any foreign key annotations although it is obvious that the class needs them. I won't change that now in order not to break anything, but I will definitely add a to-do for that.